File: src/main/java/org/example/keycloak/schemas/ScimCreateUserRequest.java
package org.example.keycloak.schemas;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record ScimCreateUserRequest(
    List<String> schemas,
    String externalId,
    ScimUserName name,
    List<ScimUserEmail> emails,
    String userName,
    boolean active
) {

  @JsonIgnoreProperties(ignoreUnknown = true)
  public record ScimUserName(
      String familyName,
      String givenName
  ) {

  }

  @JsonIgnoreProperties(ignoreUnknown = true)
  public record ScimUserEmail(
      String value,
      String type,
      boolean primary
  ) {

  }

  @JsonIgnoreProperties(ignoreUnknown = true)
  public record ScimUserMeta(
      String resourceType
  ) {

  }
}

File: src/main/kotlin/org/example/keycloak/schemas/ScimCreateUserRequest.kt
package org.example.keycloak.schemas

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class ScimCreateUserRequest(
    val schemas: List<String>,
    val externalId: String?,
    val name: ScimUserName,
    val emails: List<ScimUserEmail>,
    val userName: String,
    val active: Boolean
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ScimUserName(
        val familyName: String?,
        val givenName: String?
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ScimUserEmail(
        val value: String?,
        val type: String?,
        val primary: Boolean
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class ScimUserMeta(
        val resourceType: String?
    )
}